nginxcfg/nginxcfg.py

```diff
@@ -0,0 +1,251 @@
+        for c in remove:
+            if c in path:
+                path = path.replace(c, '')
+        return path
+
+    def _get_includes_line(self, line, parent, root):
+        """ Reads a config line, starting with 'include', and returns a list
+            of files this include corresponds to. Expands relative paths,
+            unglobs globs etc.
+        """
+        path = orig_path = self._strip_line(line.split()[1])
+        included_from_dir = os.path.dirname(parent)
+
+        if not os.path.isabs(path):
+            """ Path is relative - first check if path is
+                relative to 'current directory' """
+            path = os.path.join(included_from_dir, self._strip_line(path))
+            if not os.path.exists(os.path.dirname(path)) or \
+                    not os.path.isfile(path):
+                """ If not, it might be relative to the root """
+                path = os.path.join(root, orig_path)
+
+        if os.path.isfile(path):
+            return [path]
+        elif '/*' not in path and not os.path.exists(path):
+            """ File doesn't actually exist - probably IncludeOptional """
+            return []
+
+        """ At this point we have an absolute path to a basedir which
+            exists, which is globbed
+        """
+        basedir, extension = path.split('/*')
+        try:
+            if extension:
+                return [
+                    os.path.join(basedir, f) for f in os.listdir(
+                        basedir) if f.endswith(extension)]
+
+            return [os.path.join(basedir, f) for f in os.listdir(basedir)]
+        except OSError:
+            return []
+
+    def _get_all_config(self, config_file=None):
+        """
+        Reads all config files, starting from the main one, expands all
+        includes and returns all config in the correct order as a list.
+        """
+        config_file = config_file or "/etc/nginx/nginx.conf"
+        ret = [config_file]
+
+        with open(config_file, "r") as config_data:
+            for line in [line.strip().strip(';') for line in config_data]:
+                line = line.split('#')[0]
+                if line.startswith('include'):
+                    includes = self._get_includes_line(line,
+                                                       config_file,
+                                                       "/etc/nginx/")
+                    for include in includes:
+                        try:
+                            ret += self._get_all_config(include)
+                        except IOError:
+                            pass
+        return ret
+
+    def _get_vhosts_info(self, config_file):
+        server_block_boundry = []
+        server_block_boundry_list = []
+        vhost_data = open(config_file, "r").readlines()
+        open_brackets = 0
+        found_server_block = False
+        for line_number, line in enumerate(vhost_data):
+            if line.startswith('#'):
```

Comment:
Doesn't appear to be an issue, but this doesn't account for indented comments - it's either not needed because maybe you deal with comments later on, or it's worth adding a `line.lstrip()` here? 🤷‍♂ 
```
    #TESTING Not picked up
#TESTING correctly skipped
```